workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.ts

```diff
@@ -0,0 +1,504 @@
+    };
+
+    const response: QueryCommandOutput = await this._awsService.helpers.ddb
+      .query({
+        key
+      })
+      .execute();
+    const groupIds: string[] =
+      response.Items?.map((item) => {
+        const sk = _.get(item as Item, 'sk') as string;
+        return this._decomposeKey(sk).id;
+      }) ?? [];
+
+    return {
+      groupIds
+    };
+  }
+
+  public async getUsersFromGroup(
+    getUsersFromGroup: GetUsersFromGroupRequest
+  ): Promise<GetUsersFromGroupResponse> {
+    //Requires a GSI where the partitionKey is sk and sort key is pk
+    this._isInit();
+    const key = {
+      name: 'sk',
+      value: this._composeIdentityKey(this._assignedGroupPrefix, 'GROUP', getUsersFromGroup.groupId)
+    };
+    const response = await this._awsService.helpers.ddb
+      .query({
+        index: this._getResourceBySortKeyIndex,
+        key
+      })
+      .execute();
+    const userIds: string[] =
+      response.Items?.map((item) => {
+        const pk = _.get(item as Item, 'pk') as string;
+        return this._decomposeKey(pk).id;
+      }) ?? [];
+
+    return {
+      userIds
+    };
+  }
+
+  public async createIdentityPermissions(
+    createIdentityPermissionsRequest: CreateIdentityPermissionsRequest
+  ): Promise<CreateIdentityPermissionsResponse> {
+    this._isInit();
+    const { identityPermissions } = createIdentityPermissionsRequest;
+
+    const unprocessedIdentityPermissions: IdentityPermission[] = [];
+    const batchCreateRequestPromises = identityPermissions.map(async (identityPermission) => {
+      const condition = 'attribute_not_exists(pk)';
+      const item = this._createItemFromIdentityPermission(identityPermission);
+      try {
+        await this._awsService.helpers.ddb
+          .update(
+            {
+              pk: item.pk,
+              sk: item.sk
+            },
+            {
+              item,
+              condition
+            }
+          )
+          .execute();
+      } catch (err) {
+        //Track identity permission if unprocesssed
+        unprocessedIdentityPermissions.push(identityPermission);
+      }
+    });
+
+    //Reject
+    await Promise.all(batchCreateRequestPromises);
+    if (unprocessedIdentityPermissions.length) return { created: false, unprocessedIdentityPermissions };
+
+    return { created: true };
+  }
+
+  public async deleteIdentityPermissions(
+    deleteIdentityPermissionsRequest: DeleteIdentityPermissionsRequest
+  ): Promise<DeleteIdentityPermissionsResponse> {
+    this._isInit();
+    const { identityPermissions } = deleteIdentityPermissionsRequest;
+
+    const batchDeleteRequestPromises = _.chunk(identityPermissions, 25).map(async (identityPermissions) => {
+      const deleteKeys: { pk: string; sk: string }[] = identityPermissions.map((identityPermission) => {
+        const item = this._createItemFromIdentityPermission(identityPermission);
+        return {
+          pk: item.pk as string,
+          sk: item.sk as string
```

Comment:
do we need `as string` now that item is typed?

Reply:
abstracted away using a DBItem